doc/ref/chap-type-method.md

```diff
@@ -343,6 +343,19 @@ Similar to `keys()`, but returns the values of the dictionary.
 
 ### erase()
 
+Accepts a key and deletes the relevant key-value pair. The method is idempotent.
```

Comment:
How about saying: "erase() ensures that the given key doesn't not exist in the dictionary"

This means either deleting it, or doing nothing if it doesn't exist (which implies that erase() is idempotent)

---

FWIW I believe this same  issue is discussed in Ousterhout's book.  He regretted that for the Tcl language, they made "delete key that does not exist" an error, like it is in Python!

He thinks they should have redefined it to "ensure key does not exist", which is what we're doing here!

https://blog.pragmaticengineer.com/a-philosophy-of-software-design-review/

Reply:
The new commit has your suggested phrasing + an additional example. The docs basically repeat the tests now, but I think the extra example helps reinforce what behaviour to expect.

And A Philosophy of Software Design has just gone onto the reading list :slightly_smiling_face: